bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/ClientLibraryCategoriesDataSourceServlet.java

```diff
@@ -0,0 +1,126 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2018 Adobe Systems Incorporated
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.internal.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+
+import javax.annotation.Nonnull;
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ResourceUtil;
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+
+import com.adobe.granite.ui.clientlibs.ClientLibrary;
+import com.adobe.granite.ui.clientlibs.HtmlLibraryManager;
+import com.adobe.granite.ui.clientlibs.LibraryType;
+import com.adobe.granite.ui.components.Config;
+import com.adobe.granite.ui.components.ds.DataSource;
+import com.adobe.granite.ui.components.ds.SimpleDataSource;
+
+@Component(
+        service = { Servlet.class },
+        property = {
+                "sling.servlet.resourceTypes="+ ClientLibraryCategoriesDataSourceServlet.RESOURCE_TYPE,
+                "sling.servlet.methods=GET",
+                "sling.servlet.extensions=html"
+        }
+)
+public class ClientLibraryCategoriesDataSourceServlet extends SlingSafeMethodsServlet {
+
+    public final static String RESOURCE_TYPE = "core/wcm/components/page/v2/datasource/clientlibrarycategories";
```

Comment:
@vladbailescu - do you think this resource type makes sense? perhaps we should be a bit more general with the naming of our datasource servlets rather than tying them to a particular component + version as we currently do.

Reply:
It could even be a control that makes it into the product as a commons data source / autocomplete.

Reply:
It does make sense to have versioning IMO. I don't think there's anything wrong with a future version referencing an older one either.

Reply:
Yes, I like that approach. Would prefer `datasources` to match the naming already in the product:
`core/wcm/components/commons/datasources/clientlibrarycategories/v1`